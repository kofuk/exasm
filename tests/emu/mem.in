lui r0, 0x1

lli r1, 0x90
lui r2, 0xde
ori r2, 0xad
sw r2, (r1)
lli r2, 0
lw r2, (r1)
sw r2, (r0)
addi r0, 2

lli r2, 0xef
sbu r2, (r1)
addi r1, 1
lli r2, 0xbe
sbu r2, (r1)
addi r1, -1
lw r2, (r1)
sw r2, (r0)
addi r0, 2

lui r2, 0xbe
ori r2, 0xef
sw r2, (r1)
lbu r2, (r1)
sw r2, (r0)
addi r0, 2
addi r1, 1
lbu r2, (r1)
sw r2, (r0)
