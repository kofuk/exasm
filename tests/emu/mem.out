0xde
0xad
0xef
0xbe
0x00
0xbe
0x00
0xef
